Artifacts/windows-deploymentgroup/InstallAndConfigureDeploymentGroupAgent.ps1

```diff
@@ -144,6 +144,76 @@ function Configure-DeploymentGroupAgent
 
     if ($runAsAutoLogon)
     {
+        # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+        if ($windowsLogonAccount.Split("\").Count -eq 2)
+        {
+            $domain = $windowsLogonAccount.Split("\")[0]
+        }
+        else
+        {
+            $domain = Hostname
+        }
+        $computerName = "localhost"
+        $password = ConvertTo-SecureString $windowsLogonPassword -AsPlainText -Force
+        $credentials = New-Object System.Management.Automation.PSCredential("$domain\\$windowsLogonAccount", $password)
+        Enter-PSSession -ComputerName $computerName -Credential $credentials
+        Exit-PSSession
+
+        $ErrorActionPreference = "stop"
```

Comment:
This should be moved to the start of the script and should apply to everything, not just in this context.

Reply:
did not want to change the behavior of code outside, needed it for my try catch blocks. But moved it to the beginning after checking the rest of the code to make sure it won't cause any behavior changes elsewhere

Reply:
Moved it back here as it had a few undesired effects interms of proper error messages being displayed. The initial part of the script requires that this not be set.  After that it's fine to have this. Didn't want to touch the existing code as the scenarios are too vast for me to manually test all possible success and failure scenarios